src/loaders.rs

```diff
@@ -82,7 +83,7 @@ impl FileSystemLoader {
         &self,
         py: Python<'_>,
         template_name: &str,
-        engine: &EngineData,
+        engine: Arc<Engine>,
```

Comment:
I suppsoe the thread safety of `Arc` is required and we cannot use `Rc` that is a bit cheaper ?

Reply:
Yeah, we need thread safety because Python can send things between threads behind our back.